src/sinks/loki/sink.rs

```diff
@@ -253,7 +253,13 @@ impl EventEncoder {
         self.remove_label_fields(&mut event);
 
         let timestamp = match event.as_log().get_timestamp() {
-            Some(Value::Timestamp(ts)) => ts.timestamp_nanos_opt().expect("Timestamp out of range"),
+            Some(Value::Timestamp(ts)) => match ts.timestamp_nanos_opt() {
+                Some(timestamp) => timestamp,
+                None => {
+                    emit!(LokiEventTimestampOutOfRangeError);
+                    return None;
```

Comment:
I think we also need to use the `finalizers` here and explicitly `nack` the event via `update_status` like `finalizers.update_status(EventStatus::Errored)`. Otherwise Vector will consider them to have been successfully processed.